tt_metal/llrt/tt_cluster.hpp

```diff
@@ -97,43 +97,36 @@ class Cluster {
         vector<uint32_t> &data, uint32_t sz_in_bytes, tt_cxy_pair core, uint64_t addr, bool small_access = false) const;
 
     std::optional<std::tuple<uint32_t, uint32_t>> get_tlb_data(const tt_cxy_pair &target) const {
-        chip_id_t mmio_device_id = device_to_mmio_device_.at(target.chip);
-        tt_SiliconDevice *device =
-            dynamic_cast<tt_SiliconDevice *>(this->mmio_device_id_to_driver_.at(mmio_device_id).get());
+        tt_SiliconDevice *device = dynamic_cast<tt_SiliconDevice *>(driver_.get());
         const metal_SocDescriptor &soc_desc = this->get_soc_desc(target.chip);
         tt_cxy_pair virtual_chip_coord = soc_desc.convert_to_umd_coordinates(target);
         return device->get_tlb_data_from_target(virtual_chip_coord);
     }
 
     uint32_t get_m_dma_buf_size(chip_id_t chip_id) const {
-        chip_id_t mmio_device_id = device_to_mmio_device_.at(chip_id);
-        tt_SiliconDevice *device =
-            dynamic_cast<tt_SiliconDevice *>(this->mmio_device_id_to_driver_.at(mmio_device_id).get());
```

Comment:
not related to this change but looks like this API is not used, can you remove this